testsuite/tests/singlecluster/limitador/metrics/trlp_metrics/test_trlp_metrics.py

```diff
@@ -18,46 +22,61 @@
+            "group": "paid",
+            "user_id": paid_user_api_key.model.metadata.annotations["secret.kuadrant.io/user-id"],
+        },
+    }
+
+
+@pytest.fixture(scope="module")
+def token_usage(prometheus, pod_monitor, client, user_data):
     """Send requests to generate metrics, trigger rate limiting and return token usage"""
-    # Free user requests
-    free_tokens = 0
-    for _ in range(10):
-        response = client.post("/v1/chat/completions", auth=free_user_auth, json={**basic_request})
-        if response.status_code == 200:
-            free_tokens += response.json().get("usage", {}).get("total_tokens", 0)
-        elif response.status_code == 429:
-            break  # stop once free user is rate-limited
-
-    # Paid user requests
-    paid_tokens = 0
-    for _ in range(10):
-        response = client.post("/v1/chat/completions", auth=paid_user_auth, json={**basic_request})
-        if response.status_code == 200:
-            paid_tokens += response.json().get("usage", {}).get("total_tokens", 0)
-        elif response.status_code == 429:
-            break  # stop once paid user is rate-limited
+    usage_data = {}
+
+    for user_type, user_info in user_data.items():
+        tokens = 0
+        for _ in range(10):
+            response = client.post("/v1/chat/completions", auth=user_info["auth"], json={**basic_request})
+            if response.status_code == 200:
+                tokens += response.json().get("usage", {}).get("total_tokens", 0)
+            elif response.status_code == 429:
+                break  # stop once user is rate-limited
+
+        usage_data[f"{user_type}_total_tokens"] = tokens
 
     # Wait for metrics to propagate
     prometheus.wait_for_scrape(pod_monitor, "/metrics")
 
-    return {
-        "free_total_tokens": free_tokens,
-        "paid_total_tokens": paid_tokens,
-    }
+    return usage_data
 
 
-def test_authorized_hits_metric_exists_and_increments_free_user(
-    prometheus, limitador, route, pod_monitor, free_user_api_key, token_usage
+@pytest.mark.parametrize("user_type", USERS)
+def test_authorized_hits_metric_exists_and_increments(
+    prometheus, limitador, route, pod_monitor, user_data, token_usage, user_type
 ):
-    """Verify `authorized_hits` metric is emitted and accumulates tokens consumed for free user/group"""
+    """Verify `authorized_hits` metric is emitted and accumulates tokens consumed for users/groups"""
+    user_info = user_data[user_type]
+
     metrics = prometheus.get_metrics(
         labels={
             "pod": limitador.pod.name(),
             "limitador_namespace": f"{route.namespace()}/{route.name()}",
             "job": f"{pod_monitor.namespace()}/{pod_monitor.name()}",
-            "user": free_user_api_key.model.metadata.annotations["secret.kuadrant.io/user-id"],
-            "group": "free",
+            "user": user_info["user_id"],
+            "group": user_info["group"],
+            "model": MODEL_NAME,
```

Comment:
You added this model here, should the docstring (comment) of this test function be updated to smt like `...for users/groups/models`. If I got it right this function also validates that `model` label was present with certain value now.